npm/javy/tests/tests.js

```diff
@@ -0,0 +1,36 @@
+import { runJS, stringAsInputStream, customStringInputStream } from "./runner.js";
+
+export async function smallEcho() {
+	await runJS({
+		source: "./fixtures/small_echo.js",
+		expectedOutput: "123",
+		stdin: stringAsInputStream("123"),
+	});
+}
+
+export async function libraryEcho() {
+	const len = 16 * 1024;
+	await runJS({
+		source: "./fixtures/lib_echo.js",
+		expectedOutput: "x".repeat(len),
+		stdin: stringAsInputStream("x".repeat(len)),
+	});
+}
+
+export async function longDelay() {
+	await runJS({
+		source: "./fixtures/lib_echo.js",
+		expectedOutput: "1234",
+		stdin: customStringInputStream(async controller => {
+			for (let i = 1; i <= 4; i++) {
+				controller.enqueue(i.toString());
+				await sleep(100);
```

Comment:
Is there a different way we could try to exercise the code path that the sleep is exercising but without sleeping? I would prefer we don't introduce sleeps into the test suite. (I know CI already takes forever to run and an extra 400 ms isn't making it materially worse but I'd like to know if there's a different way forward)

Reply:
I'd be with passing a `0` instead of `100`. That way it's not synchronous but it's also the shortest delay possible. If there's a concern that the code may assume that all input is available synchronously, I think that should flush that error case out.